main/src/main/scala/sbt/Project.scala

```diff
@@ -479,7 +479,15 @@ object Project extends ProjectExtra {
     previousOnUnload(s.runExitHooks())
   }
 
-  def setProject(session: SessionSettings, structure: BuildStructure, s: State): State = {
+  def setProject(session: SessionSettings, structure: BuildStructure, s: State): State =
+    setProject(session, structure, s, st => st)
+
+  def setProject(
+      session: SessionSettings,
+      structure: BuildStructure,
+      s: State,
+      preOnLoad: State => State
+  ): State = {
```

Comment:
Not sure if instead of using two separate methods if it would ok to just use a single method with a default param (like `preOnLoad: State => State = st => st`). Not sure if that would be OK for binary compatibility.

Also if you have a better name than `preOnLoad` let me know :wink:

Reply:
Two methods is better than default params. Default params can cause binary compatibility problems so we try to avoid them in sbt.